src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Emitter.cs

```diff
@@ -64,18 +64,27 @@ public void Emit()
                 _context.AddSource($"{Identifier.BindingExtensions}.g.cs", _writer.ToSourceText());
             }
 
-            private void EmitBindCoreCall(
-                TypeSpec type,
+            private void EmitBindingLogic(
+                ComplexTypeSpec type,
                 string memberAccessExpr,
                 string configArgExpr,
                 InitializationKind initKind,
                 Action<string>? writeOnSuccess = null)
             {
-                Debug.Assert(type.CanInitialize);
-
-                if (!type.NeedsMemberBinding)
+                if (!type.HasBindableMembers)
                 {
-                    EmitObjectInit(memberAccessExpr, initKind);
+                    if (initKind is not InitializationKind.None)
```

Comment:
I'm finding that this method & related logic for "complex" type binding are too cluttered. They handle multiple edge cases which makes it hard to make a concise description.

I'll follow-up with a small refactoring PR to separate/silo the concerns. This will help me tell a better story.
- do we need to init?
- how are we assigning to the target object? (declaration for new instances returned by `Get`, or to bind to object members?)
- is this a value type (copy assignment)?
- etc.